sdk/nodejs/tests/provider/experimental/testdata/map-types/index.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright 2025-2025, Pulumi Corporation.  All rights reserved.
+
+import * as pulumi from "@pulumi/pulumi";
+
+export interface MyComponentArgs {
+    aRecordOfStrings: Record<string, string>;
+    aRecordOfNumbers?: Record<string, number>;
+    aRecordOfBooleans: Record<string, boolean>;
+    aMapOfStrings: { [key: string]: string };
+    aMapOfNumbers: { [key: string]: number };
+    aMapOfBooleans?: { [key: string]: boolean };
```

Comment:
I'm missing test cases for inputty types - the most common case that we need to support, e.g.

```
autoScalingGroupTags?: pulumi.Input<{[key: string]: pulumi.Input<string>}>;
bottlerocketSettings?: pulumi.Input<{[key: string]: any}>;
```

Overall, it's good to look at SDKs that we generate in codegen and make sure we support all use cases that appear there. E.g. the inferred schema should produce the SDK of the same shape as the original SDK, and vice versa.

The examples above are from https://github.com/pulumi/pulumi-eks/blob/master/sdk/nodejs/nodeGroupV2.ts

Reply:
We don't support the `any` type yet, that can come in a future PR. I can add a couple more tests here, but in general the inner type are all handled through recursive functions here, so I don't think we need to test them all, otherwise we end up with a combinatorial explosion of tests here, without giving us much more in terms of coverage.

Reply:
These are the primary types we need to support, so we should add tests for them. Plain types are tricky and niche, while Input is our bread and butter.

Reply:
Yeah I added tests for `Inputs`.  `any` is currently under the cut line for support in v1, so if we want that we should discuss pulling that up.

Reply:
Yes, it's fine to take `any` later :+1:

Reply:
Still missing the `pulumi.Input<{[key: string]: pulumi.Input<string>}>` case? So that the schema has no plains in it?

Reply:
Added that now, thanks!